keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,618 @@
+to a registry the cluster can access. That tool might be an IDE like VSCode,
+an infra-as-code toolchain like Pulumi, or a multi-service dev environment like Tilt.
+
+On startup, the tool should connect to the cluster and read the `local-image-loaders`
+ConfigMap. The ConfigMap embeds a versioned config.
+
+If the config specifies a registry location, the tool may automatically adjust
+image pushes to push to the specified registry.
+
+If the config specifies a `help` URL, the tool may prompt the user to set up a
+registry for faster development.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Users may see the ConfigMap and draw mistaken conclusions about how to
+  interact with it. They think deleting the ConfigMap would delete the local
+  registry (which it does not).
+
+- This KEP does not specify any automatic mechanism for keeping the
+  LocalRegistryStatus up-to-date with how the cluster is configured.
+  For example, the user might delete the registry. The cluster might
+  not have a way of knowing that the registry has died.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+None
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### The `local-image-loaders` ConfigMap
+
+Tools that configure a local registry should apply a ConfigMap to the cluster.
+
+Documentation that educates people on how to set up a registry manually should
+include instructions on what the ConfigMap should be.
+
+The name of the ConfigMap must be `local-image-loaders`.
+
+The namespace must be `kube-public`.
+
+Each field of the ConfigMap documents a way to load images into the cluster.
+This proposal only specifies one field.
+
+When a local cluster adds support for new ways to import an image that they want
+to advertise (e.g., `kind load`), they should propose new fields to the
+top-level ConfigMap.
+
+Example:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-image-loaders
+  namespace: kube-public
+```
+
+### The `localRegistryStatus.v1` field
+
+The `localRegistryStatus.v1` field of the ConfigMap specifies the registry
+details.  When specified, the field must contain YAML.
```

Comment:
exclude v1 from the description of localRegistryStatus and use `vX`.

Reply:
done